cirq/protocols/json_serialization.py

```diff
@@ -518,12 +517,12 @@ def __init__(self, obj: Any):
         # Context information and the wrapped object are stored together in
         # `object_dag` to ensure consistent serialization ordering.
         self.object_dag = []
-        context_keys = set()
+        context = []
```

Comment:
can you double check whether this could be a set?

Reply:
Making this a set requires SBKs to be hashable. In principle I agree with this, but I don't think SBK is the right level to apply the restriction; since we support serialization of non-hashable objects, SBKs should also be allowed to be non-hashable.